### PR TITLE
add basic run in docker implementation

### PR DESCRIPTION
**Pull request by d-ivashchuk**

### CODE REVIEW

File: src/docker-runner/utils.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+  // @ts-expect-error TBD
+  const argv = yargs(hideBin(process.argv)).parse() as ParsedYargs;
+
+  const args = [
+    'run',
+    '--rm',
+    '-v',
+    '$PWD:/workspace',
+    `--platform linux/amd64`,
```
Comment (d-ivashchuk) [inline src/docker-runner/utils.ts:56]: I am quite worried about this, not sure if it will not fail for non mac m1 users 🥲

Comment (chriskalmar) [reply src/docker-runner/utils.ts:56]: This needs to be evaluated based on the host OS.
Also, the images needs to be prepared for multi-platform use https://docs.docker.com/build/building/multi-platform/

File: src/docker-runner/utils.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  await downloadImageIfNotExistent({ version });
+
+  // @ts-expect-error TBD
+  const argv = yargs(hideBin(process.argv)).parse() as ParsedYargs;
+
+  const args = [
+    'run',
+    '--rm',
+    '-v',
+    '$PWD:/workspace',
```
Comment (chriskalmar) [inline src/docker-runner/utils.ts:55]: Will this be resolved correctly, or do we need to provide the actual working dir?

File: src/docker-runner/utils.ts (hunk 1 above)
Comment (d-ivashchuk) [reply src/docker-runner/utils.ts:56]: turns out it was just a warning, so this platform `linux/amd64` could be removed

File: src/docker-runner/utils.ts (hunk 2 above)
Comment (d-ivashchuk) [reply src/docker-runner/utils.ts:55]: I wrote you on slack alreayd, it's quite surprising. I think the volume is copied over correctly and config is picked up - when the process starts collecting the stories to shoot it fails for some reason 😢

File: src/docker-runner/utils.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    throw new Error('Docker is not installed');
+  }
+};
+
+const isLostPixelImageDownloaded = async ({ version }: { version: string }) => {
+  isDockerInstalled();
+  const { exitCode, stdout, stderr } = await execa('docker', [
+    'images',
+    '-q',
+    `lostpixel/lost-pixel:${version}`,
```
Comment (chriskalmar) [inline src/docker-runner/utils.ts:17]:
```suggestion
    `lostpixel/lost-pixel:v${version}`,
```

**Review by chriskalmar — CHANGES_REQUESTED** [review]

File: src/docker-runner/utils.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    await execa('docker', ['pull', `lostpixel/lost-pixel:v${version}`]);
+  }
+};
+
+type ParsedYargs = {
+  configDir: 'string';
+};
+
+export const executeDockerRun = async ({ version }: { version: string }) => {
+  await downloadImageIfNotExistent({ version });
```
Comment (chriskalmar) [inline src/docker-runner/utils.ts:42]:
```suggestion

type ParsedYargs = {
  configDir: 'string';
};

export const executeDockerRun = async ({ version }: { version: string }) => {
  isDockerInstalled();
```

**Review by chriskalmar — APPROVED** [review]

**Pull request merged**